File: solutions/typescript/secret-handshake/1/secret-handshake.ts
const commands: Record<number, string> = {
  4: "wink",
  3: "double blink",
  2: "close your eyes",
  1: "jump",
};

class Handshake {
  private binary: number[];

  constructor(decimal: number) {
    this.binary = decimal
      .toString(2)
      .split("")
      .map((n) => parseInt(n));

    if (this.binary.length !== 5) {
      const length = this.binary.length;
      for (let i = 0; i < 5 - length; i++) {
        this.binary.unshift(0);
      }
    }
  }

  commands() {
    return this.binary.reduceRight((acc: any, val: any, idx) => {
      if (val > 0) {
        if (idx in commands) {
          acc.push(commands[idx]);
        } else if (idx === 0) {
          acc.reverse();
        }
      }

      return acc;
    }, []);
  }
}

export default Handshake;

File: solutions/csharp/secret-handshake/1/SecretHandshake.cs
[Flags]
enum Action
{
    Wink = 1,
    DoubleBlink = 1 << 1,
    CloseYourEyes = 1 << 2,
    Jump = 1 << 3,
    Reverse = 1 << 4
}

internal static class ActionExtensions
{
    public static string GetString(this Action action) => action switch
    {
        Action.Wink => "wink",
        Action.DoubleBlink => "double blink",
        Action.CloseYourEyes => "close your eyes",
        Action.Jump => "jump",
        _ => throw new ArgumentOutOfRangeException(nameof(action), action, null)
    };
}


public static class SecretHandshake
{
    public static string[] Commands(int commandValue)
    {
        var flags = (Action)commandValue;
        
        var actions = Enum.GetValues<Action>()
            .Where(action => action != Action.Reverse && flags.HasFlag(action))
            .Select(action => action.GetString())
            .ToArray();

        if (flags.HasFlag(Action.Reverse))
        {
            actions = actions.Reverse().ToArray();
        }

        return actions;
    }
}
